monkestation/code/modules/mob/living/simple_animal/friendly/dog.dm

```diff
@@ -19,3 +19,42 @@
 	gold_core_spawnable = NO_SPAWN
 	unique_pet = TRUE
 	speak = list("barks!", "woofs!", "borks!", "yips!", "burks!")
+
+/mob/living/simple_animal/pet/dog/corgi/borgi
+	name = "E-N"
+	real_name = "E-N"
+	desc = "It's a borgi."
+	icon = 'monkestation/icons/mob/pets.dmi'
+	icon_state = "borgi"
+	icon_living = "borgi"
+	icon_dead = "borgi_dead"
+	var/emagged = FALSE
+	minbodytemp = 0
+	loot = list(/obj/effect/decal/cleanable/robot_debris)
+	del_on_death = TRUE
+	deathmessage = "blows apart!"
+	animal_species = /mob/living/simple_animal/pet/dog/corgi/borgi
+	nofur = TRUE
+	var/next_beep
+	var/beepsound = 'sound/items/timer.ogg'
+
+/mob/living/simple_animal/pet/dog/corgi/borgi/emag_act(mob/user)
+	if(!emagged)
+		emagged = TRUE
+		visible_message("<span class='warning'>[user] swipes a card through [src].</span>")
+		visible_message("<span class='notice'>You overload [src]s internal reactor.</span>")
+		visible_message("<span class='reallybig warning'>[src] makes an odd whining noise.</span>")
+		addtimer(CALLBACK(src, .proc/explode), 10 SECONDS)
+
+/mob/living/simple_animal/pet/dog/corgi/borgi/proc/explode()
+	explosion(get_turf(src), 0, 1, 4, 7)
+	death()
```

Comment:
There is a runtime here when the corgi explodes.

"addtimer called with a callback assigned to a qdeleted object. In the future such timers will not be supported and may refuse to run or run with a 0 wait"